frontend/src/utils/completed-courses-helpers.ts

```diff
@@ -84,83 +69,62 @@ export function getNextTerm(is_summer: boolean, classes: DNDScheduleCourse[]) {
   return classes.splice(0, counter);
 }
 
-export function getSimplifiedCourseData(
-  courses: ISimplifiedCourseData[],
-  completion: string,
-  semester: string = ""
-): ISimplifiedCourseDataAPI[] {
-  return courses.map(course => {
-    return {
-      subject: course.subject,
-      course_id: course.classId.toString(),
-      semester: semester,
-      completion: completion,
-    };
-  });
-}
-/*
-Parses a student's completed courses into a schedule with courses placed in
-whichever semester they were taken. Ignores COOP3945.
+/**
+ * Parses a student's completed courses into a schedule with courses randomly populated into the schedule.
+ * Note: we approximate "when" courses were taken by sorting by prereq order
+ * @param completedCourses the completed courses
  */
-export function parseCompletedCourses(completedCourses: ScheduleCourse[]) {
+export function parseCompletedCourses(
+  completedCourses: ScheduleCourse[]
+): { schedule: DNDSchedule; counter: number } {
   const years: number[] = [];
   const yearMap: { [key: number]: ScheduleYear } = {};
-  const termMap: { [key: string]: SeasonWord } = {
-    "10": "fall",
-    "30": "spring",
-    "40": "summer1",
-    "60": "summer2",
+
+  addBlankScheduleYear(years, yearMap, 2022);
+  yearMap[years[0]].fall.classes = [...completedCourses];
+  const [schedule, counter] = convertToDNDSchedule({ years, yearMap }, 0);
+  return { schedule, counter };
```

Comment:
Question: for this, we can actually do better. my idea is to use topological sort (implemented in json_parser) to create an ordering of courses in prereq-order. thoughts? otherwise for now, we just throw all the completed courses in fall and let the user handle moving them around.

Reply:
i think that's an awesome idea, it doesn't seem too bad to implement too! bring this up during standup and @justineluo can add it to our backlog and prioritize it!